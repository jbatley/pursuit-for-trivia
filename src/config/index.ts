export enum Modes {
  Normal = 'normal'
}

export interface Config {
  useMocks: boolean;
  sentry: string;
  lsKeys: Record<string, string>;
  api: Record<API, string>;
  mode: Record<Modes, Settings>;
}

const apiBaseURL = 'https://opentdb.com';
const localStorageKey = 'PFT.';

const config: Config = {
  useMocks: true,
  sentry: process.env.REACT_APP_SENTRY || '',
  lsKeys: {
    categories: `${localStorageKey}categories`,
    questions: `${localStorageKey}questions`,
    save: `${localStorageKey}save`,
  },
  api: {
    questionURL: `${apiBaseURL}/api.php`,
    categoryURL: `${apiBaseURL}/api_category.php`,
    tokenURL: `${apiBaseURL}/api_token.php`,
  },
  mode: {
    [Modes.Normal]: {
      maxLives: 3,
      maxTime: 30,
      difficulty: [
        { id: 1, value: 'easy' },
        { id: 2, value: 'medium' },
        { id: 2, value: 'hard' },
      ],
    },
  },
};

export const isDev = process.env.NODE_ENV === 'development';

export default config;
