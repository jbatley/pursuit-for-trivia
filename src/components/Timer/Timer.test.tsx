import React from 'react';
import {
  render, waitForDomChange, wait, act,
} from '@testing-library/react';
import config from 'config';
import Component from './Timer';

describe('Timer component', () => {
  beforeAll(() => {
    jest.clearAllMocks();
  });
  test('matches snapshot', async () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render and count down', async () => {
    config.mode.normal.maxTime = 3;
    const { getByText } = render(<Component />);
    expect(getByText('3')).toBeInTheDocument();
    await waitForDomChange();
    expect(getByText('2')).toBeInTheDocument();
  });

  test('should fire callback when timer reach 0', async () => {
    const mockFn = jest.fn(() => {});
    render(<Component cb={mockFn} maxTime={1} />);
    expect(mockFn).toHaveBeenCalledTimes(0);
    await wait(() => {
      expect(mockFn).toHaveBeenCalledTimes(1);
    });
  });

  test('should fire cb, then then resetTimer can reset', async () => {
    let toBeCalled = () => {};
    const mockFn = jest.fn((reset) => { toBeCalled = jest.fn(reset); });
    const cbFn = jest.fn();
    render(<Component maxTime={1} cb={cbFn} resetTimer={mockFn} />);
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(mockFn).toBeCalledWith(expect.any(Function));
    await wait(() => {
      expect(cbFn).toHaveBeenCalledTimes(1);
    });
    act(() => {
      toBeCalled();
      expect(toBeCalled).toHaveBeenCalledTimes(1);
    });
  });
});
