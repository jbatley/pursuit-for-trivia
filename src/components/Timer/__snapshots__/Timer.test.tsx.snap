// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component matches snapshot 1`] = `
<div
  class="container"
>
  <svg
    class="timer"
    height="75"
    viewBox="0 150 250 75"
    width="150"
  >
    <circle
      class="timer__background"
      cx="125"
      cy="125"
      r="113"
    />
    <circle
      class="timer__left"
      cx="125"
      cy="125"
      r="113"
    />
  </svg>
  <div
    class="time"
  >
    10
  </div>
</div>
`;
